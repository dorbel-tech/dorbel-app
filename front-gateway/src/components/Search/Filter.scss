@import "../sharedStyles/variables.scss";
@import "../sharedStyles/bsvars.scss";
@import "./style/nouislider.css";
@import "./style/filter.scss";
