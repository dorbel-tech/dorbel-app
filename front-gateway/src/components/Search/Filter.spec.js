'use strict';
import { mount, shallow } from 'enzyme';
import React from 'react';
import Filter from './Filter';

describe('Filter', () => {
  let props;
  let mountedFilter;

  const filter = () => {
    if (!mountedFilter) {
      mountedFilter = mount(
        <Filter.WrappedComponent {...props} />
      );
    }
    return mountedFilter;
  };

  const shallowFilter = () => shallow(<Filter.WrappedComponent {...props} />);

  const saveFilterButton = () => mountedFilter.find('#saveFilterButton');

  beforeEach(() => {
    spyOn(history, 'pushState');

    props = {
      appStore: {
        authStore: {
          isUserAdmin: jest.fn()
        },
        searchStore: {}
      },
      appProviders: {
        authProvider: {
          shouldLogin: jest.fn().mockReturnValue(true)
        },
        searchProvider: {
          search: jest.fn().mockReturnValue(Promise.resolve([])),
          resetActiveFilter: jest.fn(),
          saveFilter: jest.fn().mockReturnValue(Promise.resolve())
        },
        oheProvider: {
          loadListingEvents: jest.fn()
        }
      },
      data: {
        cities: []
      },
      client: {
        query: jest.fn().mockReturnValue(Promise.resolve())
      }
    };
    mountedFilter = undefined;
  });

  it('should initialize correctly', () => {
    const expectedFilterObj = {};

    filter();

    expect(props.client.query).not.toHaveBeenCalled();
    expect(history.pushState).toHaveBeenCalledWith(expectedFilterObj, '', jasmine.any(String));
    expect(history.pushState.calls.count()).toEqual(1);
    expect(props.appProviders.searchProvider.search).toHaveBeenCalledWith(expectedFilterObj);
    expect(props.appProviders.searchProvider.search.mock.calls.length).toEqual(1);
  });

  it('should match snapshot in initial state', () => {
    expect(shallowFilter()).toMatchSnapshot();
  });

  it('should initialize with parsed filter object', () => {
    const expectedFilterObj = {city: 2};
    spyOn(JSON, 'parse').and.returnValue(expectedFilterObj);

    filter();

    expect(props.client.query.mock.calls[0][0].variables.city_id).toBe(expectedFilterObj.city);
    expect(props.appProviders.searchProvider.search).toHaveBeenCalledWith(expectedFilterObj);
    expect(history.pushState).toHaveBeenCalledWith(expectedFilterObj, '', jasmine.any(String));
    expect(history.pushState.calls.count()).toEqual(1);
    expect(props.appProviders.searchProvider.search).toHaveBeenCalledWith(expectedFilterObj);
    expect(props.appProviders.searchProvider.search.mock.calls.length).toEqual(1);
  });

  it('should reset active filter on load', () => {
    filter();
    expect(props.appProviders.searchProvider.resetActiveFilter).toHaveBeenCalled();
  });

  it('should check for login when clicking on save filter', () => {
    filter();
    saveFilterButton().simulate('click');
    expect(props.appProviders.authProvider.shouldLogin).toHaveBeenCalled();
  });

  it('should save filter when logged in and clicking on save filter', () => {
    const mockFilter = { city: 2 };
    spyOn(JSON, 'parse').and.returnValue(mockFilter);
    props.appProviders.authProvider.shouldLogin.mockReturnValue(false);

    filter();
    saveFilterButton().simulate('click');

    expect(props.appProviders.searchProvider.saveFilter).toHaveBeenCalledWith(mockFilter);
  });

  it('should change save filter button text when filter is selected', () => {
    props.appStore.searchStore.activeFilterId = 2;
    filter();
    expect(saveFilterButton().text()).toEqual('עדכן חיפוש');
  });
});
