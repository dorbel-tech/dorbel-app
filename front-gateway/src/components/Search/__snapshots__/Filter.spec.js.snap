exports[`Filter should initialize correctly 1`] = `
<MobXStoreInjector
  appProviders={
    Object {
      "cityProvider": Object {
        "loadCities": [Function],
      },
      "oheProvider": Object {
        "loadListingEvents": [Function],
      },
      "searchProvider": Object {
        "search": [Function],
      },
    }
  }
  appStore={
    Object {
      "authStore": Object {},
      "cityStore": Object {
        "cities": Array [],
      },
    }
  }>
  <Filter
    appProviders={
      Object {
        "cityProvider": Object {
          "loadCities": [Function],
        },
        "oheProvider": Object {
          "loadListingEvents": [Function],
        },
        "searchProvider": Object {
          "search": [Function],
        },
      }
    }
    appStore={
      Object {
        "authStore": Object {},
        "cityStore": Object {
          "cities": Array [],
        },
      }
    }>
    <div>
      <div
        className="filter-toggle-container">
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}>
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button">
            סנן תוצאות
          </button>
        </Button>
      </div>
      <div
        className="filter-wrapper hideFilter">
        <div
          className="filter-city-container">
          <DropdownButton
            bsSize="large"
            className="filter-city-dropdown"
            id="cityDropdown"
            onSelect={[Function]}
            title="עיר: טוען...">
            <Uncontrolled(Dropdown)
              bsSize="large"
              id="cityDropdown"
              onSelect={[Function]}>
              <Dropdown
                bsClass="dropdown"
                bsSize="large"
                componentClass={[Function]}
                id="cityDropdown"
                onSelect={[Function]}
                onToggle={[Function]}>
                <ButtonGroup
                  block={false}
                  bsClass="btn-group"
                  bsSize="large"
                  className="dropdown"
                  justified={false}
                  vertical={false}>
                  <div
                    className="dropdown btn-group btn-group-lg">
                    <DropdownToggle
                      bsClass="dropdown-toggle"
                      bsRole="toggle"
                      bsSize="large"
                      className="filter-city-dropdown"
                      id="cityDropdown"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      open={false}
                      useAnchor={false}>
                      <Button
                        active={false}
                        aria-expanded={false}
                        aria-haspopup={true}
                        block={false}
                        bsClass="btn"
                        bsSize="large"
                        bsStyle="default"
                        className="filter-city-dropdown dropdown-toggle"
                        disabled={false}
                        id="cityDropdown"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button">
                        <button
                          aria-expanded={false}
                          aria-haspopup={true}
                          className="filter-city-dropdown dropdown-toggle btn btn-lg btn-default"
                          disabled={false}
                          id="cityDropdown"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="button"
                          type="button">
                          עיר: טוען...
                           
                          <span
                            className="caret" />
                        </button>
                      </Button>
                    </DropdownToggle>
                    <DropdownMenu
                      bsClass="dropdown-menu"
                      bsRole="menu"
                      labelledBy="cityDropdown"
                      onClose={[Function]}
                      onSelect={[Function]}
                      pullRight={false}>
                      <RootCloseWrapper
                        disabled={true}
                        event="click"
                        onRootClose={[Function]}>
                        <ul
                          aria-labelledby="cityDropdown"
                          className="dropdown-menu"
                          role="menu">
                          <MenuItem
                            bsClass="dropdown"
                            disabled={false}
                            divider={false}
                            eventKey="*"
                            header={false}
                            onKeyDown={[Function]}
                            onSelect={[Function]}>
                            <li
                              className=""
                              role="presentation">
                              <SafeAnchor
                                componentClass="a"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                role="menuitem"
                                tabIndex="-1">
                                <a
                                  href="#"
                                  onClick={[Function]}
                                  onKeyDown={[Function]}
                                  role="menuitem"
                                  tabIndex="-1">
                                  כל הערים
                                </a>
                              </SafeAnchor>
                            </li>
                          </MenuItem>
                        </ul>
                      </RootCloseWrapper>
                    </DropdownMenu>
                  </div>
                </ButtonGroup>
              </Dropdown>
            </Uncontrolled(Dropdown)>
          </DropdownButton>
        </div>
        <div
          className="sort-container filter-group-container">
          <div
            className="filter-switch-group-headersort-header">
            <b>
              סדר לפי
            </b>
          </div>
          <div
            className="sort-options">
            <div
              className="filter-input-wrapper">
              <Radio
                bsClass="radio"
                checked={true}
                disabled={false}
                inline={false}
                onChange={[Function]}
                value="publish_date">
                <div
                  className="radio">
                  <label>
                    <input
                      checked={true}
                      disabled={false}
                      onChange={[Function]}
                      type="radio"
                      value="publish_date" />
                    תאריך פרסום
                  </label>
                </div>
              </Radio>
            </div>
            <div
              className="filter-input-wrapper">
              <Radio
                bsClass="radio"
                checked={false}
                disabled={false}
                inline={false}
                onChange={[Function]}
                value="lease_start">
                <div
                  className="radio">
                  <label>
                    <input
                      checked={false}
                      disabled={false}
                      onChange={[Function]}
                      type="radio"
                      value="lease_start" />
                    תאריך כניסה
                  </label>
                </div>
              </Radio>
            </div>
          </div>
        </div>
        <div
          className="filter-group-container">
          <Checkbox
            bsClass="checkbox"
            checked={true}
            className="filter-switch-group-header"
            disabled={false}
            inline={false}
            name="roommate"
            onChange={[Function]}>
            <div
              className="filter-switch-group-header checkbox">
              <label>
                <input
                  checked={true}
                  disabled={false}
                  name="roommate"
                  onChange={[Function]}
                  type="checkbox" />
                <b>
                  הציגו לי דירות לשותפים
                </b>
              </label>
            </div>
          </Checkbox>
          <div
            className="filter-input-wrapper">
            <Checkbox
              bsClass="checkbox"
              checked={true}
              disabled={false}
              inline={false}
              name="empty"
              onChange={[Function]}>
              <div
                className="checkbox">
                <label>
                  <input
                    checked={true}
                    disabled={false}
                    name="empty"
                    onChange={[Function]}
                    type="checkbox" />
                  דירות ריקות לשותפים
                </label>
              </div>
            </Checkbox>
          </div>
          <div
            className="filter-input-wrapper">
            <Checkbox
              bsClass="checkbox"
              checked={true}
              disabled={false}
              inline={false}
              name="room"
              onChange={[Function]}>
              <div
                className="checkbox">
                <label>
                  <input
                    checked={true}
                    disabled={false}
                    name="room"
                    onChange={[Function]}
                    type="checkbox" />
                  חדר בדירת שותפים
                </label>
              </div>
            </Checkbox>
          </div>
        </div>
        <div
          className="filter-sliders-container">
          <div
            className="cost-slider">
            <h5
              className="text-center">
              טווח מחירים
            </h5>
            <Nouislider
              connect={true}
              direction="ltr"
              onChange={[Function]}
              pips={
                Object {
                  "density": 30,
                  "mode": "steps",
                }
              }
              range={
                Object {
                  "max": 7000,
                  "min": 1000,
                }
              }
              start={
                Array [
                  1000,
                  7000,
                ]
              }
              step={1000}>
              <div />
            </Nouislider>
          </div>
          <div
            className="roomsnum-slider">
            <h5
              className="text-center">
              מספר חדרים
            </h5>
            <Nouislider
              connect={true}
              direction="ltr"
              onChange={[Function]}
              pips={
                Object {
                  "density": 30,
                  "mode": "values",
                  "values": Array [
                    1,
                    2,
                    3,
                    4,
                    5,
                  ],
                }
              }
              range={
                Object {
                  "12.5%": 1.5,
                  "25%": 2,
                  "37.5%": 2.5,
                  "50%": 3,
                  "62.5%": 3.5,
                  "75%": 4,
                  "87.5%": 4.5,
                  "max": 5,
                  "min": 1,
                }
              }
              snap={true}
              start={
                Array [
                  1,
                  5,
                ]
              }>
              <div />
            </Nouislider>
          </div>
          <div
            className="size-slider">
            <h5
              className="text-center">
              גודל נכס (במ\"ר)
            </h5>
            <Nouislider
              connect={true}
              direction="ltr"
              onChange={[Function]}
              pips={
                Object {
                  "density": 30,
                  "mode": "steps",
                }
              }
              range={
                Object {
                  "20%": 40,
                  "40%": 60,
                  "60%": 80,
                  "80%": 100,
                  "max": 120,
                  "min": 26,
                }
              }
              snap={true}
              start={
                Array [
                  26,
                  120,
                ]
              }>
              <div />
            </Nouislider>
          </div>
        </div>
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={true}>
          <div
            className="container-fluid">
            <Row
              bsClass="row"
              className="filter-amenities-container"
              componentClass="div">
              <div
                className="filter-amenities-container row">
                <h5>
                  <b>
                    צמצמו את החיפוש
                  </b>
                </h5>
                <Col
                  bsClass="col"
                  componentClass="div"
                  xs={4}>
                  <div
                    className="col-xs-4">
                    <Checkbox
                      bsClass="checkbox"
                      checked={false}
                      disabled={false}
                      inline={false}
                      name="park"
                      onChange={[Function]}>
                      <div
                        className="checkbox">
                        <label>
                          <input
                            checked={false}
                            disabled={false}
                            name="park"
                            onChange={[Function]}
                            type="checkbox" />
                          חניה
                        </label>
                      </div>
                    </Checkbox>
                    <Checkbox
                      bsClass="checkbox"
                      checked={false}
                      disabled={false}
                      inline={false}
                      name="balc"
                      onChange={[Function]}>
                      <div
                        className="checkbox">
                        <label>
                          <input
                            checked={false}
                            disabled={false}
                            name="balc"
                            onChange={[Function]}
                            type="checkbox" />
                          מרפסת
                        </label>
                      </div>
                    </Checkbox>
                  </div>
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  xs={4}>
                  <div
                    className="col-xs-4">
                    <Checkbox
                      bsClass="checkbox"
                      checked={false}
                      disabled={false}
                      inline={false}
                      name="ac"
                      onChange={[Function]}>
                      <div
                        className="checkbox">
                        <label>
                          <input
                            checked={false}
                            disabled={false}
                            name="ac"
                            onChange={[Function]}
                            type="checkbox" />
                          מזגן
                        </label>
                      </div>
                    </Checkbox>
                    <Checkbox
                      bsClass="checkbox"
                      disabled={false}
                      inline={false}
                      name="ele"
                      onChange={[Function]}>
                      <div
                        className="checkbox">
                        <label>
                          <input
                            disabled={false}
                            name="ele"
                            onChange={[Function]}
                            type="checkbox" />
                          מעלית
                        </label>
                      </div>
                    </Checkbox>
                  </div>
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  xs={4}>
                  <div
                    className="col-xs-4">
                    <Checkbox
                      bsClass="checkbox"
                      checked={false}
                      disabled={false}
                      inline={false}
                      name="pet"
                      onChange={[Function]}>
                      <div
                        className="checkbox">
                        <label>
                          <input
                            checked={false}
                            disabled={false}
                            name="pet"
                            onChange={[Function]}
                            type="checkbox" />
                          מותר בע״ח
                        </label>
                      </div>
                    </Checkbox>
                    <Checkbox
                      bsClass="checkbox"
                      disabled={false}
                      inline={false}
                      name="sb"
                      onChange={[Function]}>
                      <div
                        className="checkbox">
                        <label>
                          <input
                            disabled={false}
                            name="sb"
                            onChange={[Function]}
                            type="checkbox" />
                          סורגים
                        </label>
                      </div>
                    </Checkbox>
                  </div>
                </Col>
              </div>
            </Row>
            <div
              className="filter-close">
              <div
                className="filter-close-text"
                onClick={[Function]}>
                סנן וסגור
              </div>
            </div>
          </div>
        </Grid>
      </div>
    </div>
  </Filter>
</MobXStoreInjector>
`;
